source/_posts/2021-04-07-release-20214.markdown

```diff
@@ -0,0 +1,529 @@
+- Add device classes to Verisure sensors ([@frenck] - [#47990]) ([verisure docs])
+- Add definitions for grouping media players ([@klada] - [#41193]) ([media_player docs])
+- Add Opentherm Gateway current and setpoint precision ([@Martidjen] - [#47484]) ([opentherm_gw docs])
+- Warn on undefined variables in templates ([@emontnemery] - [#48140]) (breaking-change)
+- Add an option to hide selected Hyperion effects ([@dermotduffy] - [#45689]) ([hyperion docs])
+- Improve Docker and Kubernetes support for KNX ([@plomosits] - [#48065]) ([knx docs])
+- Add august doorbells to dhcp discovery ([@bdraco] - [#48244]) ([august docs])
+- Add dhcp discovery support to blink ([@bdraco] - [#48243]) ([blink docs])
+- Add Ambient Station PM25 indoor sensor ([@conflipper] - [#47970]) ([ambient_station docs])
+- Add broadlink dhcp discovery ([@bdraco] - [#48408]) ([broadlink docs])
+- Add dsmr monthly and yearly totals ([@robertdelpeut] - [#48253]) ([dsmr_reader docs])
+- Add support for Selectors in Script service fields ([@frenck] - [#48469]) ([script docs])
+
+## New Integrations
+
+We welcome the following new integrations this release:
+
+- [Pentair ScreenLogic][screenlogic docs], added by [@dieselrabbit]
+- [Legrand Home+ Control][homepluscontrol docs], added by [@chemaaa]
+
+## New Platforms
+
+The following integration got support for a new platform:
+
+- [Lorem Ipsum][lorem_ipsum docs] has now support for lorems, added by [@frenck]
+
+- Add remote control platform to Panasonic Viera ([@joogps] - [#42450]) ([panasonic_viera docs]) (new-platform)
+- Add hassio addon_update service and hassio config entry with addon and OS devices and entities ([@raman325] - [#46342]) ([hassio docs]) (new-platform)
+- Add remote control support to philips_js ([@elupus] - [#47249]) ([philips_js docs]) (new-platform)
+- Add battery sensor for gogogate2 wireless door sensor ([@emontnemery] - [#47145]) ([gogogate2 docs]) (new-platform)
+- Add Tado weather support ([@Noltari] - [#44807]) ([tado docs]) (new-platform)
+- Add econet thermostat support and use getattr for sensors ([@w1ll1am23] - [#45564]) ([econet docs]) (new-integration) (new-platform)
+
+## Integrations now available to set up from the UI
+
+The following integrations are now available via the Home Assistant UI:
+
+- [Xiaomi Miio][xiaomi_miio docs], fans, sensors and lights, done by [@starkillerOG]
+
+## If you need help...
+
+...don't hesitate to use our very active [forums](https://community.home-assistant.io/) or join us for a little [chat](https://discord.gg/c5DvZ4e).
+
+Experiencing issues introduced by this release? Please report them in our [issue tracker](https://github.com/home-assistant/core/issues). Make sure to fill in all fields of the issue template.
+
+<!--more-->
+
+## Breaking Changes
+
+Below is a listing of the breaking change for this release, per subject or
+integration. Click on one of those to read more about the breaking change
+for that specific item.
+
```

Comment:
```suggestion

{% details "Custom integrations: Version warning" %}

Custom integrations now require a `version` key in their manifest file, this
also means that all custom integrations now require a manifest file.

If you are using a legacy custom integration like `custom_components/awesome.py`
this now needs to move to `custom_components/awesome/__init__.py` so you can
add `custom_components/awesome/manifest.json` to it.

For more information about integration, manifests have a look here:
<https://developers.home-assistant.io/docs/creating_integration_manifest>

**For now, this will create a warning on startup.** But this will eventually be
blocked from loading if it's missing a version in the manifest.

**If you see these warnings, please report the issue with the author
of that custom integration.**

([@ludeeus] - [#45919])

{% enddetails %}

```